### PR TITLE
root: handle SIGHUP and SIGUSR2, healthcheck gunicorn

### PR DESCRIPTION
**Pull request by rissson**
## Details

Extracted from #5415

---

## Checklist

-   [ ] Local tests pass (`ak test authentik/`)
-   [ ] The code has been formatted (`make lint-fix`)

If an API change has been made

-   [ ] The API schema has been updated (`make gen-build`)

If changes to the frontend have been made

-   [ ] The code has been formatted (`make web`)
-   [ ] The translation files have been updated (`make i18n-extract`)

If applicable

-   [ ] The documentation has been updated
-   [ ] The documentation has been formatted (`make website`)

### CODE REVIEW

File: internal/gounicorn/gounicorn.go
Comment (rissson) [inline internal/gounicorn/gounicorn.go:39]: Would an enum be better here?

**Review by BeryJu — APPROVED** [review]
I'd like to give this some more testing in dogfooding before we actually ship it, let's merge it after 2023.8 is out

**Comment by rissson** [comment]
I'll convert to draft to prevent accidental merging.

**Comment by rissson** [comment]
@BeryJu when merging, let's not lose the commit message from e934be42f

**Review by BeryJu — APPROVED** [review]
@rissson please have a look at the latest commit that does some minor changes, aside from that we can merge (we should probably also add documentation for this somewhere)

**Comment by rissson** [comment]
> @rissson please have a look at the latest commit that does some minor changes, aside from that we can merge (we should probably also add documentation for this somewhere)

LGTM

**Pull request merged**
